akka-http/src/main/scala/akka/http/scaladsl/server/directives/PathDirectives.scala

```diff
@@ -177,13 +177,11 @@ trait PathDirectives extends PathMatchers with ImplicitPathMatcherConstruction w
   /**
    * If the request path ends with a slash, redirect to the same uri without trailing slash in the path.
    *
-   * '''Caveat''': [[pathSingleSlash]] directive will not match inside of this directive.
```

Comment:
Or maybe it needs to be amended to talk about the root case.

Let's change the whole docs to

```
If the request path ends with a slash, redirect to the same uri without trailing slash in the path.

Note, however, that his directive doesn't apply to the uri consisting of just a single slash. HTTP does not support empty target paths, so that browsers will convert a uri such as `http://example.org` to `http://example.org/` adding
 the trailing slash. Redirecting the single slash path uri would lead to a redirection loop.

'''Caveat''': [[pathSingleSlash]] directive only match on the root path level inside of this directive.
```